hypothesis.py

```diff
@@ -0,0 +1,97 @@
+import raster
+import scipy.stats
+import math
+import pandas as pd
+
+_LONGITUDE_COLUMN_NAME = 'long'
+_LATITUDE_COLUMN_NAME = 'lat'
+_FRAUDULENT_COLUMN_NAME = 'fraud'
+
+
+def element_ttest(sample_data: pd.DataFrame,
+                  isotope_column_name: str,
+                  means_isoscape: raster.AmazonGeoTiff,
+                  variances_isoscape: raster.AmazonGeoTiff,
+                  sample_size_per_location: int,
+                  p_value_target: float):
+    '''
+    Calculates the accuracy, precision, recall based on true positives and negatives,
+    and the false positive and negatives. (go/ddf-glossary)
+
+    sample_data: pd.DataFrame with lat, long, isotope_value and fraudulent columns
+    means_isoscape: Isoscape that maps geographic coordinates to a mean isotope value.
+    variances_isoscape: Isoscape that maps geographic coordinates to the variance of
+                        isotope valuesat that location.
+    sample_size_per_location: Number of samples per geographic location used to calculate
+                              mean and variance in isoscapes.
+    p_value_target: desired p_value for the t-test (e.sample_data: 0.05)
+    '''
+    sample_data = sample_data.groupby([
+        _LONGITUDE_COLUMN_NAME,
+        _LATITUDE_COLUMN_NAME,
+        _FRAUDULENT_COLUMN_NAME])[isotope_column_name]
+
+    # Counts the number of locations in the sample have more than one row in dataset.
+    rows = 0
+
+    true_positive = 0
+    true_negative = 0
+    false_positive = 0
+    false_negative = 0
+    for group_key, isotope_values in sample_data:
+        if isotope_values.size <= 1:
+            continue
+        rows += 1
+
+        # Values from sample.
+        longitude = group_key[0]
+        latitude = group_key[1]
+        isotope_mean = isotope_values.mean()
+        isotope_variance = isotope_values.var()*(isotope_values.size /
+                                                 (isotope_values.size - 1))
+        isotope_sample_count = isotope_values.size
+
+        # Values from prediction.
+        predicted_isotope_mean = raster.get_data_at_coords(
+            means_isoscape, longitude, latitude, 0)
+        predicted_isotope_variance = raster.get_data_at_coords(
+            variances_isoscape, longitude, latitude, 0)
+        predicted_isotope_sample_count = sample_size_per_location
+
+        # t-student Test
+        _, p_value = scipy.stats.ttest_ind_from_stats(
```

Comment:
I believe Kader has code that runs the ttest on a single sample -- and we probably want to include that here or at least within ddf_common.  This way, we can be sure that what we eval against is equivalent to production.

Reply:
Sure, per this and offline conversation I've refactored and tested this.

Reply:
I'm still not clear-- where are we calling that code here?